code/modules/ruins/lavalandruin_code/puzzle.dm

```diff
@@ -206,7 +206,7 @@
 	var/icon/puzzle_icon
 
 /obj/structure/puzzle_element/Move(nloc, dir)
-	if(!isturf(nloc) ||  moving_diagonally || get_dist(get_step(src,dir),get_turf(source)) > 1)
+	if(!isturf(nloc) || get_dist(get_step(src,dir),get_turf(source)) > 1)
```

Comment:
so i don't know this part of tg code, and haven't tested if this break anything

Reply:
that puzzle requires only orthogonal movement, itd be trivial if diagonal movement was allowed